src/main/java/lotto/domain/result/Statistic.java

```diff
@@ -0,0 +1,34 @@
+package lotto.domain.result;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum Statistic {
+	THREE(3, 5000),
+	FOUR(4, 50_000),
+	FIVE(5, 1_500_000),
+	BONUS(5, 3_000_000),
+	SIX(6, 2_000_000_000);
+
+	private final int matchingNumbers;
+	private final int prize;
+
+	Statistic(int matchingNumbers, int prize) {
+		this.matchingNumbers = matchingNumbers;
+		this.prize = prize;
+	}
+
+	public static Optional<Statistic> getRank(int numberOfMatch) {
```

Comment:
https://github.com/woowacourse/java-lotto/pull/127#discussion_r382969950 에 대한 답변과 같이 남겨드립니다!

일단 당첨금에 대한 경우 Null로 처리하는 것 보다는 `당첨이 안됐다`라는 값이 이 있는 것이 좋을 것 같습니다.
Optional 또는 Null로 처리하는 경우 어디서는 `당첨이 안됐다`라는 것을 처리를 해줘야 합니다.
역할을 담당하는 곳 에서 해당 역할을 하는 것이 응집도를 높이는 방법이 되는데요.

```suggestion
	public static Statistic getRank(int numberOfMatch, boolean isBonus) {
```

와 같은 형태로 변경한다면 당첨에 대한 역할은 해당 객체에서 전부 담당할 수 있고, 이는 응집도는 높이고 결합도는 낮추는 코드가 될 수 있습니다.

추가적으로 Optional 사용 시 Optional은 Null이 오지 않는 것이 규칙이고, `isPresent`도 지양하는 것이 좋습니다.

```java
private void plusCount(Optional<Statistic> statistic) {
		Optional<RankCount> rank = null;
		if (Objects.nonNull(statistic) && statistic.isPresent()) {
			rank = gameResult.of(statistic);
		}
		if (Objects.nonNull(rank) && rank.isPresent()) {
			rank.get().plusCount();
		}
	}
```

코드를 수정한다면

```java
private void plusCount(Optional<Statistic> statistic) {
		statistic.map(gameResult::of)
			.ifPresent(RankCount::plusCount);
	}
```

와 같은 형태로 만드는 것이 가독성이 상승할 것 같습니다!